Fix CS1591 warnings

conniey opened the pull request:
I recently pulled from master and I was getting around 280+ [CS1591 ](https://msdn.microsoft.com/en-us/library/zk18c1w9.aspx?f=255&MSPPError=-2147217396) warnings: "Missing XML comment for publicly visible type or member".  

This PR:

* Fixes all those warnings by adding documentation based on documentation in [ICU4C](http://icu-project.org/apiref/icu4c/)
* Adds [Obsolete] to properties/enum values that have values that are deprecated. (ie. [Collator.HiraganaQuaternary](https://github.com/sillsdev/icu-dotnet/compare/master...conniey:fixWarnings?expand=1#diff-4edbc99f1affddcfddc80262f12a09a6R49), [Character.BINARY_LIMIT](https://github.com/sillsdev/icu-dotnet/compare/master...conniey:fixWarnings?expand=1#diff-1326ee295886b5bf056871756b1c2281R243), etc.)

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/sillsdev/icu-dotnet/29)
<!-- Reviewable:end -->

ermshiperete commented:
Wow, you were busy! Thanks!

A few minor comments on some typos I stumbled upon, and one comment with a question.

---

Reviewed 23 of 23 files at r1.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[source/icu.net/BreakIterator.cs, line 129 at r1](https://reviewable.io:443/reviews/sillsdev/icu-dotnet/29#-KdAlNfcFPpN8punMRat:-KdAlNfdDVLGUT3R8v7Z:b-foqnbg) ([raw file](https://github.com/sillsdev/icu-dotnet/blob/c36b7ca20994b5cfcacdea6f79b6a303741a34d5/source/icu.net/BreakIterator.cs#L129)):*
> for for

Also extra space after sentences

---

*[source/icu.net/BreakIterator.cs, line 139 at r1](https://reviewable.io:443/reviews/sillsdev/icu-dotnet/29#-KdAlUoyirjYRLm4K96O:-KdAlUoyirjYRLm4K96P:bpo95d1) ([raw file](https://github.com/sillsdev/icu-dotnet/blob/c36b7ca20994b5cfcacdea6f79b6a303741a34d5/source/icu.net/BreakIterator.cs#L139)):*
> for for

Also first word is missing the `T`

---

*[source/icu.net/Character.cs, line 76 at r1](https://reviewable.io:443/reviews/sillsdev/icu-dotnet/29#-KdAmFjHqXSduC7tf9d-:-KdAmFjHqXSduC7tf9d0:bxtuilk) ([raw file](https://github.com/sillsdev/icu-dotnet/blob/c36b7ca20994b5cfcacdea6f79b6a303741a34d5/source/icu.net/Character.cs#L76)):*
> words,plus

missing space

---

*[source/icu.net/UnicodeString.cs, line 31 at r1](https://reviewable.io:443/reviews/sillsdev/icu-dotnet/29#-KdAoWbEQqFztt3H__0z:-KdAoWbEQqFztt3H__1-:b57fry9) ([raw file](https://github.com/sillsdev/icu-dotnet/blob/c36b7ca20994b5cfcacdea6f79b6a303741a34d5/source/icu.net/UnicodeString.cs#L31)):*
> this

replace with: `<paramref name="src"/>`, or simply add `string`

---

*[source/icu.net/UnicodeString.cs, line 85 at r1](https://reviewable.io:443/reviews/sillsdev/icu-dotnet/29#-KdAowo3740l7dES14-_:-KdAowo44n6uXDOzNBec:boqsiwb) ([raw file](https://github.com/sillsdev/icu-dotnet/blob/c36b7ca20994b5cfcacdea6f79b6a303741a34d5/source/icu.net/UnicodeString.cs#L85)):*
> this

see above

---

*[source/icu.net/UnicodeString.cs, line 142 at r1](https://reviewable.io:443/reviews/sillsdev/icu-dotnet/29#-KdAp0pcgFNG0Y5Hnq6q:-KdAp0pcgFNG0Y5Hnq6r:b-7k3uui) ([raw file](https://github.com/sillsdev/icu-dotnet/blob/c36b7ca20994b5cfcacdea6f79b6a303741a34d5/source/icu.net/UnicodeString.cs#L142)):*
> titlecased.It

Missing space

---

*[source/icu.net/Collation/HiraganaQuaternary.cs, line 11 at r1](https://reviewable.io:443/reviews/sillsdev/icu-dotnet/29#-KdApqzRcuzlnCBHWtmt:-KdApqzRcuzlnCBHWtmu:b-45t2iv) ([raw file](https://github.com/sillsdev/icu-dotnet/blob/c36b7ca20994b5cfcacdea6f79b6a303741a34d5/source/icu.net/Collation/HiraganaQuaternary.cs#L11)):*
> ```Smalltalk
> 	/// sort order
> 	/// </summary>
> 	public enum HiraganaQuaternary
> ```

With the `HiraganaQuaternary` property marked as obsolete I wonder if this enum should also be obsoleted? Or is this used somewhere else?

---

*[source/icu.net/Collation/HiraganaQuaternary.cs, line 18 at r1](https://reviewable.io:443/reviews/sillsdev/icu-dotnet/29#-KdApfEdighrN9yuVRIh:-KdApfEdighrN9yuVRIi:b-tt53m8) ([raw file](https://github.com/sillsdev/icu-dotnet/blob/c36b7ca20994b5cfcacdea6f79b6a303741a34d5/source/icu.net/Collation/HiraganaQuaternary.cs#L18)):*
> position positions

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/icu-dotnet/29#-:-KdAqMbC3d-BttWg5DnX:b-qdzb26)*
<!-- Sent from Reviewable.io -->

conniey commented:
Thanks for looking through the PR. :)

---

Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[source/icu.net/BreakIterator.cs, line 129 at r1](https://reviewable.io:443/reviews/sillsdev/icu-dotnet/29#-KdAlNfcFPpN8punMRat:-KdC_oMod7BzQ76yu34H:bcnbp59) ([raw file](https://github.com/sillsdev/icu-dotnet/blob/c36b7ca20994b5cfcacdea6f79b6a303741a34d5/source/icu.net/BreakIterator.cs#L129)):*
>  for for sentences 
Fixed.

---

*[source/icu.net/BreakIterator.cs, line 139 at r1](https://reviewable.io:443/reviews/sillsdev/icu-dotnet/29#-KdAlUoyirjYRLm4K96O:-KdC_r7JvpkAkdExGSis:bt2vr1b) ([raw file](https://github.com/sillsdev/icu-dotnet/blob/c36b7ca20994b5cfcacdea6f79b6a303741a34d5/source/icu.net/BreakIterator.cs#L139)):*
<details><summary><i>Previously, ermshiperete (Eberhard Beilharz) wrote…</i></summary><blockquote>

> for for

Also first word is missing the `T`
</blockquote></details>

Fixed.

---

*[source/icu.net/Character.cs, line 76 at r1](https://reviewable.io:443/reviews/sillsdev/icu-dotnet/29#-KdAmFjHqXSduC7tf9d-:-KdCaVwjzI3xQUmIjw5t:b47jvgo) ([raw file](https://github.com/sillsdev/icu-dotnet/blob/c36b7ca20994b5cfcacdea6f79b6a303741a34d5/source/icu.net/Character.cs#L76)):*
> etween pieces of words,plus
Fixed

---

*[source/icu.net/UnicodeString.cs, line 31 at r1](https://reviewable.io:443/reviews/sillsdev/icu-dotnet/29#-KdAoWbEQqFztt3H__0z:-KdCapFqEq8Ezwl7KIsX:b-qac6y) ([raw file](https://github.com/sillsdev/icu-dotnet/blob/c36b7ca20994b5cfcacdea6f79b6a303741a34d5/source/icu.net/UnicodeString.cs#L31)):*
> Convert the characters in this to lower case following the
Fixed

---

*[source/icu.net/UnicodeString.cs, line 85 at r1](https://reviewable.io:443/reviews/sillsdev/icu-dotnet/29#-KdAowo3740l7dES14-_:-KdCb7kbs_faNaiyxvOE:bhknshd) ([raw file](https://github.com/sillsdev/icu-dotnet/blob/c36b7ca20994b5cfcacdea6f79b6a303741a34d5/source/icu.net/UnicodeString.cs#L85)):*
> Convert the characters in this to UPPER CASE 
Fixed

---

*[source/icu.net/UnicodeString.cs, line 142 at r1](https://reviewable.io:443/reviews/sillsdev/icu-dotnet/29#-KdAp0pcgFNG0Y5Hnq6q:-KdCbCwIIY_h5S1HguIH:bhb59cw) ([raw file](https://github.com/sillsdev/icu-dotnet/blob/c36b7ca20994b5cfcacdea6f79b6a303741a34d5/source/icu.net/UnicodeString.cs#L142)):*
> titlecased.It
Fixed

---

*[source/icu.net/Collation/HiraganaQuaternary.cs, line 11 at r1](https://reviewable.io:443/reviews/sillsdev/icu-dotnet/29#-KdApqzRcuzlnCBHWtmt:-KdCbF6bTblxW0s7kjCV:b-3qz0f) ([raw file](https://github.com/sillsdev/icu-dotnet/blob/c36b7ca20994b5cfcacdea6f79b6a303741a34d5/source/icu.net/Collation/HiraganaQuaternary.cs#L11)):*
<details><summary><i>Previously, ermshiperete (Eberhard Beilharz) wrote…</i></summary><blockquote>

With the `HiraganaQuaternary` property marked as obsolete I wonder if this enum should also be obsoleted? Or is this used somewhere else?
</blockquote></details>

It's not used anywhere else. I added it. :)

---

*[source/icu.net/Collation/HiraganaQuaternary.cs, line 18 at r1](https://reviewable.io:443/reviews/sillsdev/icu-dotnet/29#-KdApfEdighrN9yuVRIh:-KdCbfvV_TIy_XvW8V1C:b9yqx2c) ([raw file](https://github.com/sillsdev/icu-dotnet/blob/c36b7ca20994b5cfcacdea6f79b6a303741a34d5/source/icu.net/Collation/HiraganaQuaternary.cs#L18)):*
> position positions
Fixed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/icu-dotnet/29#-:-KdCbiQnAScumo_DWBTG:b-mmhvv1)*
<!-- Sent from Reviewable.io -->

ermshiperete commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 4 of 4 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/icu-dotnet/29#-:-KdPcMy82jn49JZZRG8D:bnfp4nl)*
<!-- Sent from Reviewable.io -->
